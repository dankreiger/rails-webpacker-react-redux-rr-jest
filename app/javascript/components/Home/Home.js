import React from 'react';
import { HomeContainer } from './Home.styles';

const Home = () => {
  return (
    <HomeContainer>
      Starter with RoR, React, Redux, React-Router, RSpec, and Jest
    </HomeContainer>
  );
};

export default Home;
