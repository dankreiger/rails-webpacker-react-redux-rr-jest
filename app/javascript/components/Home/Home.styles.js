import styled from 'styled-components';
import { Container } from 'reactstrap';

export const HomeContainer = styled(Container)`
  height: 100vh;
  display: flex;
  align-items: center;
  justify-content: center;
`;
