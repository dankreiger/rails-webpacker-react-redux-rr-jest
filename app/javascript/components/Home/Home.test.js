import Home from './Home';
import { shallow } from 'enzyme';

describe('Home', () => {
  let homeComponent;

  beforeEach(() => {
    homeComponent = shallow(<Home />);
  });

  describe('rendering', () => {
    test('renders as expected', () => {
      expect(homeComponent.length).toBe(1);
      expect(homeComponent).toMatchSnapshot();
    });
  });
});
