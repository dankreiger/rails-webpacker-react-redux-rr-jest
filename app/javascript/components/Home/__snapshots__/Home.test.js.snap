// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home rendering renders as expected 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Home />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": "Starter with RoR, React, Redux, React-Router, RSpec, and Jest",
    },
    "ref": null,
    "rendered": "Starter with RoR, React, Redux, React-Router, RSpec, and Jest",
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": true,
        "rules": Array [
          "
  height: 100vh;
  display: flex;
  align-items: center;
  justify-content: center;
",
        ],
      },
      "displayName": "Styled(Container)",
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": [Function],
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
      Symbol(Symbol.hasInstance): [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": "Starter with RoR, React, Redux, React-Router, RSpec, and Jest",
      },
      "ref": null,
      "rendered": "Starter with RoR, React, Redux, React-Router, RSpec, and Jest",
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": true,
          "rules": Array [
            "
  height: 100vh;
  display: flex;
  align-items: center;
  justify-content: center;
",
          ],
        },
        "displayName": "Styled(Container)",
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": [Function],
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
        Symbol(Symbol.hasInstance): [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
